examples/context/theme-detailed-app.js

```diff
@@ -0,0 +1,46 @@
+import {ThemeContext, themes} from './theme-context';
+import ThemedButton from './button';
+
+// An intermediate component that uses the ThemedButton
+const Toolbar = props => {
+  return (
+    <ThemedButton onClick={props.changeTheme}>
+      Change Theme
+    </ThemedButton>
+  );
+};
+
+class App extends React.Component {
+  state = {
+    theme: themes.light,
+  };
+
+  toggleTheme = () => {
+    this.setState(state => ({
+      theme:
+        state.theme === themes.dark
+          ? themes.light
+          : themes.dark,
+    }));
+  };
+
+  render() {
+    //highlight-range{1-3}
+    // The ThemedButton button inside the ThemeProvider
+    // uses the theme from state while the one outside uses
+    // the default dark theme
+    //highlight-range{3-5,7}
+    return (
+      <div>
```

Comment:
Can we come up with some semantic elements here? I understand they're not very relevant but React apps often gets criticized for using divs everywhere, and I'd prefer we don't invite this criticism with our own doc examples.

Reply:
```jsx
  <React.Fragment>
    <ThemeContext.Provider value={this.state.theme}>
      <Toolbar changeTheme={this.toggleTheme} />
        </ThemeContext.Provider>
    <Contents>
      <ThemedButton />
    </Contents>
  </React.Fragment>
```

?